src/pytorch_lightning/strategies/horovod.py

```diff
@@ -129,27 +133,33 @@ def _unpack_lightning_optimizer(opt):
             # Synchronization will be performed explicitly following backward()
             self._exit_stack.enter_context(optimizer.skip_synchronize())
 
-    def barrier(self, *args, **kwargs):
+    def barrier(self, *args: Any, **kwargs: Any) -> None:
         if distributed_available():
             self.join()
 
-    def broadcast(self, obj: object, src: int = 0) -> object:
+    def broadcast(self, obj: TBroadcast, src: int = 0) -> TBroadcast:
         obj = hvd.broadcast_object(obj, src)
         return obj
 
-    def model_to_device(self):
+    def model_to_device(self) -> None:
         if self.root_device.type == "cuda":
             # this can potentially be removed after #8312. Not done due to lack of horovod testing
             torch.cuda.set_device(self.root_device)
+        assert self.model is not None
         self.model.to(self.root_device)
 
-    def join(self):
+    def join(self) -> None:
         if self.root_device.type == "cuda":
             hvd.join(self.local_rank)
         else:
             hvd.join()
 
-    def reduce(self, tensor, group: Optional[Any] = None, reduce_op: Optional[Union[ReduceOp, str]] = "mean"):
+    def reduce(
+        self,
+        tensor: Union[Any, Tensor],
```

Comment:
yes `Union[Any, Tensor]` -> `Any` but it is confusing as `tensor` argument can be any

Reply:
I left a similar comment in a PR quite a while ago, but I remember we concluded that we'd continue to have types like this instead of merging them to `Any`. (cc @justusschock if I remember it correctly)

Reply:
The advantage that way is that actually most IDEs nowadays parse the typehints and that way you do get autocompletion for tensors which in many cases is the correct type instead of just doing Any, where the IDE doesn't know what to suggest. So yeah, technically this is redundant, but it really helps with delevoper experience

Reply:
I am in favor of having Any, Tensor